projects/packages/my-jetpack/_inc/state/reducers.js

```diff
@@ -6,10 +6,54 @@ import { combineReducers } from '@wordpress/data';
 /**
  * Internal dependencies
  */
-import { SET_PURCHASES, SET_PURCHASES_IS_FETCHING } from './actions';
+import {
+	SET_PURCHASES,
+	SET_PURCHASES_IS_FETCHING,
+	SET_PRODUCT_ACTION_ERROR,
+	SET_PRODUCT_STATUS,
+	IS_FETCHING_PRODUCT_STATUS,
+} from './actions';
 
-const products = ( state = {} ) => {
-	return state;
+const products = ( state = {}, action ) => {
+	switch ( action.type ) {
+		case SET_PRODUCT_STATUS: {
+			const { productId, status } = action;
+			return {
+				...state,
+				items: {
+					...state.items,
+					[ productId ]: {
+						...state.items[ productId ],
+						...status,
+					},
+				},
+				error: {},
```

Comment:
The place to reset the error is not `IS_FETCHING_PRODUCT_STATUS` then here? Since we're doing a fetch or re-fetch we should clear the error, right?